src/js/components/ExportButton.js

```diff
@@ -0,0 +1,80 @@
+/* @flow */
+import {useDispatch} from "react-redux"
+import React, {useState} from "react"
+import styled from "styled-components"
+
+import {remote} from "electron"
+
+import ExportIcon from "../icons/ExportIcon"
+import InfoNotice from "./InfoNotice"
+import MacSpinner from "./MacSpinner"
+import ToolbarButton from "./ToolbarButton"
+import XButton from "./XButton"
+import exportResults from "../flows/exportResults"
+import useIpcListener from "./hooks/useIpcListener"
+import useSetTimeout from "./hooks/useSetTimeout"
+
+const SpinnerWrap = styled.div`
+  transform: scale(0.8);
+  margin-right: 12px;
+`
+
+const ExportingMessage = () => (
+  <>
+    <p>Exporting...</p>
+    <SpinnerWrap>
+      <MacSpinner light />
+    </SpinnerWrap>
+  </>
+)
+
+const DoneMessage = ({onClick}) => (
+  <>
+    <p>Export complete.</p>
+    <XButton onClick={onClick} />
+  </>
+)
+
+function showDialog() {
+  return remote.dialog.showSaveDialog(remote.getCurrentWindow(), {
+    title: "Export Results as ZNG",
+    buttonLabel: "Export",
+    defaultPath: "results.zng",
+    properties: ["createDirectory"]
+  })
+}
+
+export default function ExportButton() {
+  let dispatch = useDispatch()
+  let [status, setStatus] = useState("INIT")
+  let setTimeout = useSetTimeout()
+
+  useIpcListener("exportResults", onClick, [])
```

Comment:
It seems more maintainable to put the ipc listeners in the components that care about them, rather than in the initializer `initIpcListeners.js` file.

Reply:
Makes sense to me